### DISCUSSION TITLE
How to split a very large multi-page tiff file?

### DISCUSSION THREAD

**Discussion by seitay** [ImageMagick/ImageMagick#4463] [Help]
Hello,

I have a very large multi-page tiff file which is 245GB in size and includes 1600+ pages(slices).  I have succeeded in splitting the image by allocating about 1TB RAM on the high performance computing platform.  However, it takes up quite a RAM and not ideal in the competing situation for allocating the memory nor running on a less powerful machine with something like 32GB RAM.  I was wondering if ImageMagick provides a CLI feature to read page by page instead of the entire file in a RAM and writes out the each split one by one.  Do you support such a feature?  Also, if you suggest a better way to tackle this conversion, I would appreciate your advise.  

Thank you,
Seita

---
This is the command used for splitting.
```
convert multipage.tif -set filename:f "%[t]_%[fx:t+1]" +adjoin "%[filename:f].tif"
```

**Comment by snibgo** [comment]
For IM v7, I suggest "magick" instead of "convert".

How about:
```
magick multipage.tif[0] out-0.tif
magick multipage.tif[1] out-1.tif
magick multipage.tif[2] out-2.tif
```
... and so on.

Alternatively, a common utility is "tiffsplit":
```
tiffsplit multipage.tif out
```

**Comment by seitay** [comment]
Thank you @snibgo , I would like to try upgrading IM to v7 and use those utilities.  Let me give that a try.

**Comment by snibgo** [comment]
If you have v6, use "convert" instead of "magick".

"tiffsplit" is not part of IM.

**Comment by seitay** [comment]
Hi again @snibgo What would be the advantage of using "magick" and "tiffsplit" over "convert"?

**Comment by snibgo** [comment]
The command for current IM is "magick". The old command was "convert".

IM still includes "convert" for backwards compatibility, for old scripts that rely on the old v6 syntax. The current v7 syntax is more powerful, and has better error checking.

I suggest tiffsplit as a possible alternative. I am not suggesting magick _and_ tiffsplit.

**Reply by seitay** [reply]
tiffsplit is also a nice utility.  I wish it has the flexibility to change the ouput suffix in numerical order.  It seems I can only get it in the range [aaa-zzz].  "magick" and "convert" work great. Thanks for the discussion.

**Reply by valdier1** [reply]
Do you have a link for tiffsplit? searching for this in google gives a ton of ambiguous results

**Comment by snibgo** [comment]
tiffsplit is available for Windows from the Cygwin package manager.

```
f:\web\im>tiffsplit
LIBTIFF, Version 4.3.0
Copyright (c) 1988-1996 Sam Leffler
Copyright (c) 1991-1996 Silicon Graphics, Inc.

Split a multi-image TIFF into single-image TIFF files

usage: tiffsplit input.tif [prefix]
```
